internal/shim-sgx/src/handler/mod.rs

```diff
@@ -126,59 +315,131 @@ impl<'a> Handler<'a> {
             Some(ExceptionVector::Page) => {
                 h.print_ssa_stack_trace();
                 h.gdb_session();
-                h.exit(1)
+                let _ = h.exit_group(1);
+                unreachable!()
             }
 
             _ => h.attacked(),
         }
     }
 
+    fn get_attestation(
+        &mut self,
+        hash: usize,
+        hash_len: usize,
+        buf: usize,
+        buf_len: usize,
+    ) -> Result<[usize; 2], libc::c_int> {
+        if buf == 0 {
+            return Ok([QUOTE_SIZE, TECH]);
+        }
+
+        if buf_len > isize::MAX as usize {
+            return Err(libc::EINVAL);
+        }
+
+        if buf_len < QUOTE_SIZE {
+            return Err(libc::EMSGSIZE);
+        }
+
+        if hash_len != 64 {
+            return Err(libc::EINVAL);
+        }
```

Comment:
This wall of `if`s could be easier to digest if it was in a single `match`

Reply:
Something like this? hmm.. I don't know..

```rust
        match (hash_len, buf, buf_len) {
            (_, 0, _) => return Ok([QUOTE_SIZE, TECH]),
            (_, _, buf_len) if buf_len > isize::MAX as usize => return Err(libc::EINVAL),
            (_, _, buf_len) if buf_len < QUOTE_SIZE => return Err(libc::EMSGSIZE),
            (hash_len, _, _) if hash_len != 64 => return Err(libc::EINVAL),
            _ => {}
        }
```